src/utils/generator.js

```diff
@@ -280,10 +282,17 @@ class PossiblyNormalReturnEntry extends GeneratorEntry {
     super();
     this.completion = completion;
     this.containingGenerator = generator;
-
     this.condition = completion.joinCondition;
-    this.consequentGenerator = Generator.fromEffects(completion.consequentEffects, realm, "ConsequentEffects");
-    this.alternateGenerator = Generator.fromEffects(completion.alternateEffects, realm, "AlternateEffects");
+
+    // The effects of the normal path have already been applied to generator
+    let empty_effects = construct_empty_effects(realm);
+    empty_effects[0] = completion.value;
+    let consequentEffects =
+      completion.consequent instanceof AbruptCompletion ? completion.consequentEffects : empty_effects;
```

Comment:
Are all of the normal effects already applied or just the generator (specifically thinking of modified bindings/properties/created objects)?

Reply:
That is my understanding.